### PR TITLE
Set/get in update track

### PR DESCRIPTION
**Pull request by guperrot**
Please have a look at our [guidelines for contributions](https://github.com/microsoft/appcenter-sdk-android/blob/develop/CONTRIBUTING.md) and consider the following before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you add unit tests?
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?
* [ ] Did you check UI tests on the sample app? They are not executed on CI.

## Description

See [AB#75285](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/75285)

### CODE REVIEW

File: sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/UpdateTrack.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+package com.microsoft.appcenter.distribute;
+
+/**
+ * Track to use for in-app updates.
+ */
+public enum UpdateTrack {
```
Comment (MatkovIvan) [inline sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/UpdateTrack.java]: Let's use just int with annotations (like in other places) verification to easier wrap it into another languages

Comment (guperrot) [reply sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/UpdateTrack.java]: True, though annotations were not supposed to be public as we depend on a package for that. We can use annotations only for internal code, not for the public signature.

Comment (guperrot) [reply sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/UpdateTrack.java]: We discussed some spec changes anyway and will include the change to int to the PR.

Comment (guperrot) [reply sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/UpdateTrack.java]: Actually we return that value with a getter so it's not a good thing for the customer. It's ok if it makes bindings slightly more difficult, it's more important that the API is easy to use in Java.

File: sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/Distribute.java [hunk 2]
```diff
@@ -452,10 +487,17 @@ public synchronized void onActivityCreated(Activity activity, Bundle savedInstan
...
-            }
+            resetWorkflow();
+        }
+    }
+
+    /**
+     * Reset current state to allow a new update check.
+     */
+    private void resetWorkflow() {
+        if (mChannel != null && getStoredDownloadState() == DOWNLOAD_STATE_COMPLETED) {
```
Comment (guperrot) [inline sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/Distribute.java]: This kind of check is what prevents the UI from switching to the new value @clpolet. Resume is checking for that too. I tested in app with a timer and got:

> 2020-01-21 16:40:35.053 9757-9757/com.microsoft.appcenter.sasquatch.project D/AppCenterDistribute: Previous dialog is still being shown in the same activity.
> 2020-01-21 16:40:35.053 9757-9757/com.microsoft.appcenter.sasquatch.project V/AppCenterDistribute: Already checking or checked latest release.

Looks like we are good.

**Comment by guperrot** [comment]
Ran into problems with the old agent pool that was used for CI (also affecting develop). Migrated to the new azure pipelines setting to fix.

**Comment by guperrot** [comment]
Verified lint locally with the modification and it passes, strange that it appears only now.

File: apps/sasquatch/src/main/java/com/microsoft/appcenter/sasquatch/activities/EventActivity.java [hunk 3]
```diff
@@ -188,7 +188,7 @@ private boolean onlyStringProperties() {
     }
 
     @Override
-    protected void send(View view) {
+    public void send(View view) {
```
Comment (guperrot) [inline apps/sasquatch/src/main/java/com/microsoft/appcenter/sasquatch/activities/EventActivity.java]: old VM could not run tests anymore for some reason (even on develop: https://msmobilecenter.visualstudio.com/SDK/_build/results?buildId=833380).
Switch to another pool and got the new lint error, which means lint behavior depends on what is installed on the machine for some reason.

**Pull request merged**